File: backend/app/domain/entities/staff.py
from __future__ import annotations

from datetime import datetime
from enum import Enum
from uuid import UUID

from pydantic import BaseModel


class StaffRole(str, Enum):
    """Role classification for staff members."""

    HOUSEKEEPER = "HOUSEKEEPER"
    PROPERTY_MANAGER = "PROPERTY_MANAGER"


class StaffMember(BaseModel):
    """A member of the operational team (housekeeper or property manager)."""

    id: UUID
    first_name: str
    last_name: str
    email: str
    phone: str | None = None
    role: StaffRole
    is_active: bool = True
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/staff.ts
export type StaffRole = 'housekeeper' | 'property_manager';

export type AssignmentStatus = 'scheduled' | 'completed' | 'cancelled';

export interface StaffMember {
  id: string;
  first_name: string;
  last_name: string;
  email: string;
  phone: string | null;
  role: StaffRole;
  is_active: boolean;
  property_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface HousekeepingAssignment {
  id: string;
  reservation_id: string;
  staff_id: string;
  staff_name: string;
  property_name?: string;
  guest_name?: string;
  scheduled_date: string;
  notes: string | null;
  status: AssignmentStatus;
  created_at: string;
  updated_at: string;
}
